scripts/complete_index.py

```diff
@@ -0,0 +1,68 @@
+"""
+Script to identify and complete general index entries with original content.
+"""
+
+import glob
+import sys
+
+import polib
+
+
+def complete_index(po_files=None):
+    """
+    Identifies general index entries based on source order and completes them
+    with original content.
+
+    args:
+        po_files: List of .po files to process. If not provided, it processes
+        all .po files in the current directory and immediate subdirectories.
+
+    returns:
+        str: Files and entries modified.
+    """
+
+    # Read .po files
+    if not po_files:
+        po_files = [f for f in glob.glob("**/*.po", recursive=True)]
+
+    modified_texts = []
+    po_entries = []
+    for file in po_files:
+        try:
+            po_file = polib.pofile(file)
+            po_entries = [entry for entry in po_file if not entry.obsolete]
+            val_max = 0
+            marks = []
+
+            # Compare source index with file order and create marks
+            for i, entry in enumerate(po_entries):
+
+                source_index = int(entry.occurrences[0][1])
+                if source_index <= val_max:
+                    marks.append(i)
```

Comment:
Si sigues la sugerencia de mover la lógica a una nueva función, entonces, como digo en otro comentario, ésta puede ser un generador:

```suggestion
                    yield entry
```

Nota que eso también significa que no necesitas el `enumerate` en el `for` de arriba